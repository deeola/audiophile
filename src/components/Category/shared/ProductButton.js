import React from 'react'
import ProductDetails from './ProductDetails';
import {Button} from '../../Buttons/Button'
import { ProductDetailsGeneral } from './SharedStyling';

const ProductButton = ({link, label, title, np, subtitle}) => {
  return (
    <ProductDetailsGeneral data-testid='productbutton' >
        <ProductDetails np={np} subtitle={subtitle} title={title} />
        <Button datatestid={`productbutton-${title}`} label={label} link={link} />
    </ProductDetailsGeneral>
  )
}

export default ProductButton
