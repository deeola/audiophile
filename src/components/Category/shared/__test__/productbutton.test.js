import { render } from "@testing-library/react";
import reactDom from "react-dom";
import renderer from "react-test-renderer";

import { BrowserRouter } from "react-router-dom";
import ProductButton from "../ProductButton";

it("renders product button div correctly", () => {
  const div = document.createElement("div");
  reactDom.render(
    <BrowserRouter>
      <ProductButton />
    </BrowserRouter>,
    div
  );
});

it("renders 2 components", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <ProductButton />
    </BrowserRouter>
  );

  expect(getByTestId("productbutton").childElementCount).toBe(2);
});


// snapshot test

it('matches snapshot', () => {
    const tree = renderer.create(
        <BrowserRouter>
        <ProductButton />
      </BrowserRouter>
        
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
    })


