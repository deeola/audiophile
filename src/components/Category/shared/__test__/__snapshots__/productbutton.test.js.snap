// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<article
  className="sc-hKMtZM bQLcuJ"
  data-testid="productbutton"
>
  <div
    className="sc-eCYdqJ bMkDvt"
  >
    <p
      className="sc-iBkjds dIztKO"
      data-testid="np"
    >
       
    </p>
    <h2
      className="sc-jSMfEi imCpJA"
      data-testid="title"
    />
    <p
      className="sc-gKXOVf dgecyH"
      data-testid="subtitle"
    />
  </div>
  <a
    className="sc-ftvSup KzLBC"
    color="primary"
    data-testid="productbutton-undefined"
    hover="primary"
    href=""
    onClick={[Function]}
  />
</article>
`;
